src/main/webapp/app/grading-system/grading-system.component.ts

```diff
@@ -613,4 +631,137 @@ export class GradingSystemComponent implements OnInit {
             gradeType: GradeType.GRADE,
         };
     }
+
+    /**
+     * Triggered when csv file is uploaded
+     * @param event file read event
+     */
+    async onCSVFileSelect(event: any) {
+        if (event.target.files.length > 0) {
+            await this.readGradingStepsFromCSVFile(event, event.target.files[0]);
+            this.lowerBoundInclusivity = true;
+            this.setInclusivity(this.gradingScale.gradeSteps);
+            this.maxPoints = 100;
+            this.onChangeMaxPoints(this.maxPoints);
+            this.determineFirstPassingGrade();
+            this.gradingScale.gradeSteps.sort((a, b) => a.lowerBoundPercentage - b.lowerBoundPercentage);
+        }
+    }
+
+    /**
+     * Import grade steps from csv file
+     * @param event the read event
+     * @param csvFile the csv file
+     * @private
+     */
+    async readGradingStepsFromCSVFile(event: any, csvFile: File) {
+        let csvGradeSteps: CsvGradeStep[] = [];
+        try {
+            csvGradeSteps = await this.parseCSVFile(csvFile);
+        } catch (error) {
+            return [];
+        }
+
+        if (csvGradeSteps.length === 0) {
+            this.gradingScale.gradeSteps = [];
+            return;
+        }
+
+        // parse and set the grade type
+        const gradeType = csvGradeSteps[0]['bonusPoints'] === undefined ? GradeType.GRADE : GradeType.BONUS;
+        if (gradeType === GradeType.BONUS) {
+            this.gradingScale.gradeType = GradeType.BONUS;
+        } else {
+            this.gradingScale.gradeType = GradeType.GRADE;
+        }
+
+        this.gradingScale.gradeSteps = this.mapCsvGradeStepsToGradeSteps(csvGradeSteps, gradeType);
+    }
+
+    /**
+     * Map the imported csv objects to GradeStep object
+     * In case that the grade type of the imported file is bonus, attribute gradeName will be set with the bonusPoints attribute
+     * @param csvGradeSteps
+     */
+    mapCsvGradeStepsToGradeSteps(csvGradeSteps: CsvGradeStep[], gradeType: GradeType): GradeStep[] {
+        return csvGradeSteps.map(
+            (csvGradeStep) =>
+                ({
+                    gradeName:
+                        gradeType === GradeType.GRADE
+                            ? csvGradeStep[csvColumnsGrade.gradeName]
+                                ? String(csvGradeStep[csvColumnsGrade.gradeName])
+                                : ''
+                            : csvGradeStep[csvColumnsBonus.bonusPoints]
+                            ? String(csvGradeStep[csvColumnsBonus.bonusPoints])
+                            : '',
```

Comment:
The triple ternary operator makes this very hard to read and understand what is going on.

Reply:
Resolved in commit 588e8c1.